src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        }
+
+        /// <summary>
+        /// Set the Ina219 ADC resolution or samples to be used when reading the Bus voltage.
+        /// </summary>
+        /// <remarks>
+        /// This can either by the number of bits used for the ADC conversion (9-12 bits) or the number of samples at 12 bits to be averaged for the result.
+        /// </remarks>
+        /// <param name="resorsamp">The ADC resolution or sample count to switch to.</param>
+        public void SetBusAdcResolutionOrSamples(AdcResolutionOrSamples resorsamp)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.BadcMask ^ 0xFFFF;
+            regValue |= (ushort)((ushort)resorsamp << 4);
+
+            WriteRegister(Register.Configuration, regValue);
+
+            _busAdcResSamp = resorsamp;
+        }
+
+        /// <summary>
+        /// Set the Ina219 ADC resolution or samples to be used when reading the Shunt voltage.
+        /// </summary>
+        /// <remarks>
+        /// This can either by the number of bits used for the ADC conversion (9-12 bits) or the number of samples at 12 bits to be averaged for the result.
+        /// </remarks>
+        /// <param name="resorsamp">The ADC resolution or sample count to switch to.</param>
+        public void SetShuntAdcResolutionOrSamples(AdcResolutionOrSamples resorsamp)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.SadcMask ^ 0xFFFF;
+            regValue |= (ushort)resorsamp;
+
+            WriteRegister(Register.Configuration, regValue);
+
+            _shuntAdcResSamp = resorsamp;
+        }
+
+        /// <summary>
+        /// Set the Ina219 calibration value used to scale the Shunt voltage into a Current reading.
+        /// </summary>
+        /// <remarks>
+        /// If this method is used then the values for the CurrentLsb and PowerMsb will have to be manually determined
+        /// to ensure accurate readings from GetPower and GetCurrent Methods.
+        /// <see cref="GetPower"/><seealso cref="GetCurrent"/><seealso cref="PowerLsb"/><seealso cref="CurrentLsb"/>
+        /// </remarks>
+        /// <param name="calibrationValue">The calibration value to switch to.</param>
+        public void SetCalibration(ushort calibrationValue)
+        {
+            _calibrationValue = calibrationValue;
+            WriteRegister(Register.Calibration, calibrationValue);
+        }
+
+        /// <summary>
+        /// Set the Ina219 calibration values used to scale the Shunt voltage into a Current reading based on the sensitivity of the
+        /// Progmamable Gain Amplifier, the required full scale current and the value of the shunt resistor.
+        /// </summary>
+        /// <remarks>
+        /// This method sets the calibration register and also the PowerLsb and CurrentLsb properties.
```

Comment:
this should explain what will change from user perspective and when should people use it and how and not implementation detail